Add `format.timeFormatter` to get raw access to the formatter

cibernox opened the pull request:
The reason behind this PR is that I need to call `formatter.formatToParts` instead of simply `formatter.format`.

I'm happy to change the approach if you think some other approach is more flexible while allowing to call `formatToParts`.

If this approach seems ok I can add the same method for the dateFormatter too.

Fixes #30

kaisermann commented:
Hey! Thanks for the PR.

What do you think about exporting the `getXxxxxFormatter` methods? 

```ts
// index.ts
export {
  getDateFormatter,
  getNumberFormatter,
  getTimeFormatter,
  getMessageFormatter,
} from './includes/formatters'
```

Then you would be able to pass a locale and all the other parameters you want.

cibernox commented:
@kaisermann I'm ok with that, but unless I'm mistaken if the user changes the selected locale, the app wouldn't be notified the same way they are when using store, right?

For completenes, this is a store I've created that returns a function  that knows how to format the time of the next high/low tide

```js
import { t } from "svelte-i18n";

export const formatNextTideDate = derived([t], ([format]) => {
  return (nextTideDate, centerDate) => {
    let centerDateInSeconds = centerDate.getTime() / 1000;
    let nextTideDateInSeconds = nextTideDate.getTime() / 1000;
    if (
      centerDateInSeconds >= nextTideDateInSeconds - FIFTEEN_MINUTES_IN_SECONDS &&
      centerDateInSeconds <= nextTideDateInSeconds + FIFTEEN_MINUTES_IN_SECONDS
    ) {
      return format("tide.now");
    } else {
      return format.time(nextTideDate, { format: 'short' });
    }
  }
});
```

kaisermann commented:
Yeah, being a low-level API, you would need to keep the `locale` updated. Following your example, it's simple to do that, just also derive from the `locale` store:

```js
import { t, locale } from '...'

export const customFormatThingie = derived([t, locale], (format, currentLocale) => ....);
```

Would that help you?

cibernox commented:
@kaisermann yeah, that makes sense. I've updated the PR to export the `getXxxxxFormatter`s from the index. I've updated `test/client/includes/formatters.test.ts` to import the tested utilities from the index, so we're also asserting not only that they work but that they are exposed publicly.

cibernox commented:
Any feedback on this one that I should address?

cibernox commented:
ping

kaisermann commented:
Hey, @cibernox, I'm going to take a look at this PR's code soon. Been busy with the beginning of the year 😁

kaisermann commented:
Released on `v2.2.0`. Added some documentation to the wiki: https://github.com/kaisermann/svelte-i18n/wiki/Methods#low-level-api

cibernox commented:
Great, thanks!
